systemtest/src/test/java/io/strimzi/systemtest/specific/SpecificST.java

```diff
@@ -100,20 +110,151 @@ void testRackAware() {
+                .build();
+
+        internalKafkaClient.verifyProducedAndConsumedMessages(
+                internalKafkaClient.sendMessagesPlain(),
+                internalKafkaClient.receiveMessagesPlain()
         );
     }
 
+    @Test
+    @Tag(CONNECT)
+    @Tag(REGRESSION)
+    void testRackAwareConnectWrongDeployment() throws Exception {
+        installClusterOperator(NAMESPACE, CO_OPERATION_TIMEOUT_SHORT);
+
+        String wrongRackKey = "wrong-key";
+        String rackKey = "rack-key";
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3)
+                .editSpec()
+                    .editKafka()
+                        .withNewRack()
+                            .withTopologyKey(rackKey)
+                        .endRack()
+                        .addToConfig("replica.selector.class", "org.apache.kafka.common.replica.RackAwareReplicaSelector")
+                    .endKafka()
+                .endSpec().done();
+
+        KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME).done();
+        String kafkaClientsPodName = kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME).get(0).getMetadata().getName();
+
+        LOGGER.info("Deploy KafkaConnect with wrong rack-aware topology key: {}", wrongRackKey);
+        KafkaConnectResource.kafkaConnectWithoutWait(KafkaConnectResource.defaultKafkaConnect(CLUSTER_NAME, CLUSTER_NAME, 1)
+                .editSpec()
+                    .withNewRack()
+                        .withTopologyKey(wrongRackKey)
+                    .endRack()
+                .addToConfig("key.converter.schemas.enable", false)
+                .addToConfig("value.converter.schemas.enable", false)
+                .addToConfig("key.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .addToConfig("value.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .endSpec().build());
+
+        PodUtils.waitForPendingPod(CLUSTER_NAME + "-connect");
+        List<String> connectWrongPods = kubeClient().listPodNames(Labels.STRIMZI_KIND_LABEL, KafkaConnect.RESOURCE_KIND);
+        String connectWrongPodName = connectWrongPods.get(0);
+        LOGGER.info("Waiting for ClusterOperator to get timeout operation of incorrectly set up KafkaConnect");
+        TestUtils.waitFor("Wait for KafkaConnect 'Unschedulable' condition due to wrong rackKey used.",
+                Constants.GLOBAL_POLL_INTERVAL, CO_OPERATION_TIMEOUT_SHORT * 2, () -> {
+                List<Condition> conditions = KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus().getConditions();
+                for (Condition condition : conditions) {
+                    if (condition.getReason().matches("TimeoutException") && condition.getType().matches("NotReady")) {
+                        return true;
+                    }
+                }
+                return false;
+            });
```

Comment:
Move it to some utils class? You basically wait until KC is not ready, I don't see any specific message in status. Also, the test will wait 5 minutes until deployment will timeout. WOuldn't be enough just to wait for the pending, wait for the next reconciliation, check if the pod is still pending, and then proceed?

Reply:
Makes sense to move it to `KCUtils.waitUntilCondition(Reason, Type)`. Good idea.

Actually it will wait only 1min as `long CO_OPERATION_TIMEOUT_SHORT = Duration.ofSeconds(30).toMillis();` *2

Reply:
I see. Anyway, move it is still a good idea :)